### PR TITLE
chore(sdk): Support artifact types under google namespace

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
Preserve type names in `schemaTitle` for types under `google` namespace.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: sdk/python/kfp/v2/components/types/type_utils.py [hunk 1]
```diff
@@ -33,6 +33,9 @@
     'markdown': artifact_types.Markdown,
 }
 
+_GOOGLE_TYPES_PATTERN = r'^google.[A-Za-z]+$'
+_GOOGLE_TYPES_VERSION = '0.0.1'
```
Comment (IronPan) [inline sdk/python/kfp/v2/components/types/type_utils.py:37]: not blocking - what would be the long term plan for specifying the artifact types in YAML?

**Review by IronPan — COMMENTED** [review]
/lgtm

File: sdk/python/kfp/v2/components/types/type_utils.py (hunk 1 above)
Comment (chensun) [reply sdk/python/kfp/v2/components/types/type_utils.py:37]: I'm thinking about a dictionary like type: `{google.VertexModel: {version: 0.0.1}}`

**Pull request merged**